### ISSUE TITLE
Training the whole FSANET_Capsule for new dataset

### ISSUE THREAD

**Issue by sainikhilrai** [shamangary/FSA-Net#14]
Hi,
I am able to run all the pre-trained models and tested with my new image too. I got pretty good result. I want to use the same network architecture of FSANET_Capsule (for e.g.) for training with new data set. Since FSANET_Capsule is composed of other modules too like ssr_G_model, ssr_S_model, ssr_Cap_model and ssr_F_Cap_model.
Now, I have the following doubts with respect to training with new dataset:
1. How the training should be done?
2. Is my understanding about the training part code (FSANET_train.py) in the directory 'FSA-Net/training_and_testing', using the pre-trained weights for each of different modules are correct? This leads to my next doubt.
3. Do we have to do training for all the modules? If "yes" what are the hyper-parameters and loss function for each? It is not mentioned in the paper clearly and the repository too. It would be really helpful if you could tell what those each modules are doing so that it would be easy to train and choose appropriate loss function for them?
4. How the training of the each module should be done? Do we have to do training of each module one after the another or together (combine)?

Thanks in advance.

**Comment by shamangary** [comment]
Hello @sainikhilrai,

The training is not that complex. It is done by using a simple L1 loss (MAE loss). The modules are not trained separately, everything is trained all together for FSA-Net. There is no loading pre-trained weight for the training or separate module training. 1 loss for all!

The only thing you need to remember is that FSA_net_Capsule, FSA_net_Var_Capsule, and  FSA_net_noS_Capsule are trained individually. This is corresponding to the different scoring function in the paper.

Modules:
1. ssr_G_model_build:
https://github.com/shamangary/FSA-Net/blob/master/training_and_testing/FSANET_model.py#L672
Two-stream structure for extracting the features.

2. ssr_feat_S_model_build:
https://github.com/shamangary/FSA-Net/blob/master/training_and_testing/FSANET_model.py#L765
Generating fine-grained structure mapping from different scoring functions.

3. ssr_S_model_build:
https://github.com/shamangary/FSA-Net/blob/master/training_and_testing/FSANET_model.py#L778
Apply the mapping on to the features and generate primary capsules.

4. ssr_Cap_model_build:
https://github.com/shamangary/FSA-Net/blob/master/training_and_testing/FSANET_model.py#L820
Feed the primary capsules into capsule layer and output the final aggregated capsule features.
And divide them into 3 parts.

5. ssr_F_model_build:
https://github.com/shamangary/FSA-Net/blob/master/training_and_testing/FSANET_model.py#L846
Taking the previous 3 part features for Soft-stagewise regression (SSR) module.
Including prediction, dynamic index shifting, and dynamic scaling.
This part please check the '[IJCAI18] SSR-Net' for more detail explanation.

6. SSR_module:
https://github.com/shamangary/FSA-Net/blob/master/training_and_testing/FSANET_model.py#L891
Taking the prediction, dynamic index shifting, and dynamic scaling for the final regression output.
In this case, there are three outputs (yaw, pitch, roll).

**Comment by sainikhilrai** [comment]
Thanks for the nice explanation about each stages. I will do the training part with my new data set.
